### PR TITLE
remove helper function for unused storage feature in pkg/proxy/util

### LINKED ISSUE #118120: helper function for unused storage feature in pkg/proxy/util
`pkg/proxy/util/utils.go` (which is supposed to be utility functions for kube-proxy) contains the function `NewFilteredDialContext` and associated struct `FilteredDialogOptions`, which are not used by kube-proxy and have nothing to do with service proxying. They were added in #91785 as part of a storage-related feature.

However this feature was apparently only ever implemented for two storage backends (scaleio and storageos), both of which have since been removed (#101685, #111620). So the current state is that you can set the associated [options in the kube-controller-manager config v1alpha1 API](https://github.com/kubernetes/kubernetes/pull/91785/commits/f2e23afcf13fef805ad5341bd7a5625de5982d6c#diff-5ad0cddac612ee34ef81af71efaeae956cc08c468cf20ac2bf719aba980a73a6), and kcm constructs a `proxyutil.FilteredDialogOptions` which then gets passed around the volume controller, but there is no code that ever actually _uses_ that object.

I would like to get this code out of `pkg/proxy/util`, because it doesn't belong there. I was originally going to suggest moving it to `pkg/volume/util` or something, but given that it's essentially unused, maybe we should just revert #91785?

/sig storage

### PR DESCRIPTION
**Pull request by carlory**
#### What type of PR is this?

/kind cleanup
/kind deprecation

#### What this PR does / why we need it:

See https://github.com/kubernetes/kubernetes/issues/118120#issuecomment-1553796570

#### Which issue(s) this PR fixes:

Fixes #118120

#### Does this PR introduce a user-facing change?

```release-note
kube-controller-manager deprecate `--volume-host-cidr-denylist` and `--volume-host-allow-local-loopback` flags.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/proxy/util/nodeport_addresses_test.go [hunk 1]
```diff
@@ -21,7 +21,7 @@ import (
 	"net"
 	"testing"
 
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
```
Comment (danwinship) [inline pkg/proxy/util/nodeport_addresses_test.go:24]: unnecessary change to an otherwise unmodified file

**Comment by danwinship** [comment]
/approve
for the `pkg/proxy/util` parts. But I don't know if this is the path sig-storage wants; discussion on #118120 was not resolved

**Comment by mattcary** [comment]
Let me see if I can prod some people into commenting on #118120.

**Comment by mattcary** [comment]
Thanks for the ping; we've resolved the conversation in #118120 and agree with removing the unused code.

/lgtm

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Comment by thockin** [comment]
@danwinship - one more reason to move pkg/proxy under cmd/kube-proxy and/or staging/src/k8s.io/kube-proxy :)

**Comment by pacoxu** [comment]
only rebased
/lgtm
